shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -590,58 +586,57 @@ - (void)setDisplayingFlutterUI:(BOOL)displayingFlutterUI {
 
 - (void)callViewRenderedCallback {
   self.displayingFlutterUI = YES;
-  if (_flutterViewRenderedCallback != nil) {
-    _flutterViewRenderedCallback.get()();
-    _flutterViewRenderedCallback.reset();
+  if (self.flutterViewRenderedCallback) {
+    self.flutterViewRenderedCallback();
+    self.flutterViewRenderedCallback = nil;
   }
 }
 
 - (void)removeSplashScreenView:(dispatch_block_t _Nullable)onComplete {
-  NSAssert(_splashScreenView, @"The splash screen view must not be null");
-  UIView* splashScreen = [_splashScreenView.get() retain];
-  _splashScreenView.reset();
+  NSAssert(self.splashScreenView, @"The splash screen view must not be nil");
+  UIView* splashScreen = self.splashScreenView;
+  _splashScreenView = nil;
```

Comment:
`self.splashScreenView = nil` throws you into an infinite loop, hence the ivar.

Reply:
Probably worth a brief comment in the code.

Reply:
Done.